test/extensions/quic_listeners/quiche/envoy_quic_server_stream_test.cc

```diff
@@ -368,5 +368,80 @@ TEST_P(EnvoyQuicServerStreamTest, WatermarkSendBuffer) {
   EXPECT_TRUE(quic_stream_->write_side_closed());
 }
 
+TEST_P(EnvoyQuicServerStreamTest, HeadersContributeToWatermarkIquic) {
```

Comment:
Do we have a time frame for moving off off gQUIC?
Mainly asking because there's a lot of nearly copy pasting and I wonder if it's worth refactoring for shared code or if it's going away soon enough it's not worth the TODO.

Reply:
~1 year to deprecate gQUIC code. It's hard to share the code between server and client stream/session because they already inherits from different base classes.